src/utils/helpers.js

```diff
@@ -0,0 +1,23 @@
+export function renderLoading(isLoading, button, defaultText='Save' , loadingText="Saving. . .") {
+  if (isLoading) {
+    button.textContent = loadingText;
+  } else {
+    button.textContent = defaultText;
+  }
+}
+
+export function handleSubmit(request, evt, loadingText = 'Saving...') {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  const initialText = submitButton.textContent;
+  renderLoading(true, submitButton, initialText, loadingText);
+  request()
+    .then(() => {
+      // evt.target.reset();
```

Comment:
You can add `disableButton` here to disable the save button after resetting the form